File: smartadapter/src/main/java/smartadapter/widget/AutoRemoveItemSwipeExtension.java
package smartadapter.widget;

/*
 * Created by Manne Öhlund on 2019-08-17.
 * Copyright (c) All rights reserved.
 */

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import smartadapter.SmartRecyclerAdapter;
import smartadapter.viewholder.SmartAdapterHolder;

/**
 * Automatically removes an item in {@link SmartRecyclerAdapter} when swiped.
 *
 * @see BasicSwipeExtension
 * @see SmartAdapterHolder
 */
public class AutoRemoveItemSwipeExtension extends BasicSwipeExtension implements SmartAdapterHolder {

    private SmartRecyclerAdapter smartRecyclerAdapter;

    @Override
    public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction) {
         super.onSwiped(viewHolder, direction);
         smartRecyclerAdapter.removeItem(viewHolder.getAdapterPosition());
    }

    @Override
    public void setSmartRecyclerAdapter(@NonNull SmartRecyclerAdapter smartRecyclerAdapter) {
        this.smartRecyclerAdapter = smartRecyclerAdapter;
    }
}

File: smartadapter/src/main/java/smartadapter/widget/AutoRemoveItemSwipeExtension.kt
package smartadapter.widget

/*
 * Created by Manne Öhlund on 2019-08-17.
 * Copyright (c) All rights reserved.
 */

import androidx.recyclerview.widget.RecyclerView

import smartadapter.SmartRecyclerAdapter
import smartadapter.viewholder.SmartAdapterHolder

/**
 * Automatically removes an item in [SmartRecyclerAdapter] when swiped.
 *
 * @see BasicSwipeExtension
 *
 * @see SmartAdapterHolder
 */
class AutoRemoveItemSwipeExtension : BasicSwipeExtension(), SmartAdapterHolder {

    override lateinit var smartRecyclerAdapter: SmartRecyclerAdapter

    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
        super.onSwiped(viewHolder, direction)
        smartRecyclerAdapter.removeItem(viewHolder.adapterPosition)
    }
}
